src/app/manager-dashboard/reports/reports-myplanet.component.ts

```diff
@@ -42,6 +42,15 @@ export class ReportsMyPlanetComponent implements OnInit {
   versions: string[] = [];
   selectedVersion = '';
   disableShowAllTime = true;
+  selectedTimeFilter = 'all'; // Default to All Time
+  timeFilterOptions = [
+    { value: '24h', label: $localize`Last 24 Hours` },
+    { value: '7d', label: $localize`Last 7 Days` },
+    { value: '30d', label: $localize`Last 30 Days` },
+    { value: 'all', label: $localize`All Time` },
+    { value: 'custom', label: $localize`Custom` },
+  ];

```

Comment:
Since we have a similar block of code in the myPlanet logs, it would be ideal to place it in a constanst.ts file and import to both myPlanet logs & reports